frontend/src/pages/question/panels/NodePanel.jsx

```diff
@@ -151,22 +150,77 @@ export default function NodePanel({ panelStore }) {
     // so that they show up in the list in the
     // right order
     const combinedConcepts = conceptsFormatted.map(
-      (c, i) => [c, conceptsSetified[i]]).flat();
+      (c, i) => [c, conceptsSetified[i]],
+    ).flat();
 
     node.setConcepts(combinedConcepts);
   }
   // When node panel mounts get concepts
   useEffect(() => { updateConceptList(); }, [biolink]);
 
   async function fetchCuries(newSearchTerm) {
-    // Get and update list of curies
-    const response = await API.ranker.entityLookup(newSearchTerm);
-    if (response.status === 'error' || !_.isArray(response)) {
+    // Get list of curies that match this search term
+    const response = await API.nameResolver.entityLookup(newSearchTerm, 1000);
+    if (response.status === 'error') {
       displayAlert('error',
-        'Failed to contact server to search curies. You will still be able to select generic types. Please try again later');
-    } else {
-      node.updateCuries(response);
+        'Failed to contact name resolver to search curies. You will still be able to select generic types. Please try again later');
+      node.setLoading(false);
+      return;
+    }
+    const curies = Object.keys(response);
+
+    // Pass curies to nodeNormalizer to get type information and
+    // a better curie identifier
+    //
+    // Right now curies are in the URL so we need to split it so we don't
+    // hit maximum URL length of the server
+
+    // Based on experimentation, the max URL length is around 7000 characters
+    // Higher values are faster, so we want to set this as high as possible
+    const maxUrlLength = 7000;
+
+    // Iterate over each curie chunking based on the length that the URL will be
+    const curiesChunked = [[]];
+    curies.forEach((c) => {
+      const existingArray = curiesChunked[curiesChunked.length - 1];
+      const existingLength = existingArray.join('&curie=').length;
+      const newLength = existingLength + c.length + '&curie='.length;
+      if (newLength < maxUrlLength) {
+        existingArray.push(c);
+      } else {
+        curiesChunked.push([c]);
+      }
+    });
+
+    const normalizationAPICallResponses = await Promise.all(
+      curiesChunked.map((cs) => API.nodeNormalization.getNormalizedNodes(cs)),
+    );
+
+    // Fail if there are any errors
+    let nodeNormalizerError = null;
+    normalizationAPICallResponses.forEach((r) => {
+      if (r.status === 'error') {
+        nodeNormalizerError = r;
+      }
+    });
+
+    if (nodeNormalizerError === 'error') {
+      displayAlert('error',
+        'Failed to contact node normalizer to search curies. You will still be able to select generic types. Please try again later');
+      node.setLoading(false);
+      return;
```

Comment:
This seems like it won't work. We're setting nodeNormalizerError to the whole response object if its status is 'error', then checking if that object is equal to the string 'error'. It won't. The forEach can also be a find() so we stop as soon as we get an error. Cause on line 211, we stop and return if there are any errors. Are there any chances one call to the normalizer will fail while others work? Do we want to handle that case, or just show an error if any of them fail?